onyo/lib/inventory.py

```diff
@@ -190,6 +191,29 @@ def _get_pending_asset_names(self) -> list[str]:
                 names.append(op.operands[1].name)
         return names
 
+    def _get_pending_dirs(self) -> list[Path]:
+        """Gets inventory dirs that would come into existence due to pending operations.
+
+        Extracts paths to inventory dirs, that are the anticipated results of pending
+        moves and creations.
+
+        Notes
+        -----
+        Currently used within `rename_directory` to allow for move+rename.
+        This needs enhancement/generalization (check for removed ones as well, etc.)
+
+        Returns
+        -------
+        list of Path
+            Inventory dirs about to be created.
+        """
+        dirs = []
+        for op in self.operations:
+            if op.operator == OPERATIONS_MAPPING['new_directories']:
+                dirs.append(op.operands[1])
```

Comment:
It looks to me like the tests never go into this "if" case. It should not be hard or time consuming to add a test for this inventory function, too. :)

Alternatively, if this is more work than I think, it can be addressed with https://github.com/psyinfra/onyo/issues/501.

Reply:
Yes, unused ATM. #546 is the spot to rework and test this. :)